main/src/observables/gravitational_waves.hpp

```diff
@@ -77,69 +77,157 @@ std::array<T, 8> gravRad(Dataset& d,size_t first, size_t last, T viewTheta, T vi
+                                             d.x.data(),
+                                             d.y.data(),
+                                             d.z.data(),
+                                             d.vx.data(),
+                                             d.vy.data(),
+                                             d.vz.data(),
+                                             d.ax.data(),
+                                             d.ay.data(),
+                                             d.az.data(),
+                                             d.m.data());
+    d2Q_local[Q::xz] = d2QuadpoleMomentum<T>(first,
+                                             last,
+                                             Dim::x,
+                                             Dim::z,
+                                             d.x.data(),
+                                             d.y.data(),
+                                             d.z.data(),
+                                             d.vx.data(),
+                                             d.vy.data(),
+                                             d.vz.data(),
+                                             d.ax.data(),
+                                             d.ay.data(),
+                                             d.az.data(),
+                                             d.m.data());
+    d2Q_local[Q::yz] = d2QuadpoleMomentum<T>(first,
+                                             last,
+                                             Dim::y,
+                                             Dim::z,
+                                             d.x.data(),
+                                             d.y.data(),
+                                             d.z.data(),
+                                             d.vx.data(),
+                                             d.vy.data(),
+                                             d.vz.data(),
+                                             d.ax.data(),
+                                             d.ay.data(),
+                                             d.az.data(),
+                                             d.m.data());
+
+    int              rootRank = 0;
     std::array<T, 6> d2Qxx_global;
-    MPI_Reduce(d2Q_local.data(), d2Qxx_global.data(),6, MpiType<T>{}, MPI_SUM, rootRank, MPI_COMM_WORLD);
+    MPI_Reduce(d2Q_local.data(), d2Qxx_global.data(), 6, MpiType<T>{}, MPI_SUM, rootRank, MPI_COMM_WORLD);
 
     T httplus;
     T httcross;
 
     computeHtt(d2Qxx_global, viewTheta, viewPhi, &httplus, &httcross);
 
-    return {httplus, httcross, d2Qxx_global[0], d2Qxx_global[1], d2Qxx_global[2], d2Qxx_global[3], d2Qxx_global[4], d2Qxx_global[5]};
+    return {httplus,
+            httcross,
+            d2Qxx_global[0],
+            d2Qxx_global[1],
+            d2Qxx_global[2],
+            d2Qxx_global[3],
+            d2Qxx_global[4],
+            d2Qxx_global[5]};
 }
```

Comment:
This is very difficult to follow. d2Q_global should be implemented in a similar way as d2Q_local, where we can specify dimension coordinates, and no array indexes.

Reply:
This should be already done / fixed, might be left-over due to missing merge.